governance/program/src/instruction.rs

```diff
@@ -0,0 +1,247 @@
+    ///
+    ///   0. `[writable]` Initialized Signatory account
+    ///   1. `[writable]` Initialized Signatory Mint account
+    ///   2. `[signer]` Admin account
+    ///   3. `[writable]` Proposal State account
+    ///   4. `[]` Proposal account
+    ///   5. '[]` Token program account
+    AddSignatory,
+
+    /// [Requires Admin token]
+    /// Removes a Signatory from the Proposal
+    ///
+    ///   0. `[writable]` Signatory account to remove token from
+    ///   1. `[writable]` Signatory Mint account
+    ///   2. `[signer]` Admin account
+    ///   3. `[writable]` Proposal State account
+    ///   4. `[]` Proposal account
+    ///   5. `[signer]` Transfer authority
+    ///   6. '[]` Token program account
+    RemoveSignatory,
+
+    /// [Requires Admin token]
+    /// Adds an instruction to the Proposal. Max of 5 of any  type. More than 5 will throw error
+    ///
+    ///   0. `[writable]` Uninitialized Proposal SingleSignerInstruction account
+    ///   1. `[writable]` Proposal state account
+    ///   2. `[signer]` Admin account
+    ///   3. `[]` Proposal account
+    ///   4. `[]` Governance account
+    AddSingleSignerInstruction {
+        /// Slot waiting time between vote period ending and this being eligible for execution
+        hold_up_time: u64,
+
+        /// Instruction
+        instruction: Instruction,
+
+        /// Position in instruction array
+        position: u8,
+    },
+
+    /// [Requires Admin token]
+    /// Remove instruction from the Proposal
+    ///
+    ///   0. `[writable]` Proposal State account
+    ///   1. `[writable]` Proposal SingleSignerInstruction account
+    ///   2. `[signer]` Admin account
+    ///   3. `[]` Proposal account
+    RemoveInstruction,
+
+    /// [Requires Admin token]
+    /// Update instruction hold up time in the Proposal
+    ///
+    ///   0. `[writable]` Proposal SingleSignerInstruction account
+    ///   1. `[signer]` Admin account
+    ///   2. `[]` Proposal State account
+    ///   3. `[]` Proposal account
+    UpdateInstructionHoldUpTime {
+        /// Minimum waiting time in slots for an instruction to be executed after proposal is voted on
+        hold_up_time: u64,
+    },
+
+    /// [Requires Admin token]
+    /// Cancels Proposal and moves it into Canceled
+    ///
+    ///   0. `[writable]` Proposal state account
+    ///   1. `[writable]` Admin account
+    ///   2. `[]` Proposal account
+    CancelProposal,
+
+    /// [Requires Signatory token]
+    /// Burns signatory token, indicating you approve of moving this Proposal from Draft state to Voting state
+    /// The last Signatory token to be burned moves the state to Voting
+    ///
+    ///   0. `[writable]` Proposal State account
+    ///   1. `[writable]` Signatory account
+    ///   2. `[writable]` Signatory Mint account
+    ///   3. `[]` Proposal account
+    ///   4. `[signer]` Transfer authority
+    ///   5. `[]` Token program account
+    ///   6. `[]` Clock sysvar
+    SignProposal,
```

Comment:
The usage of "Sign" here confused me for a moment.  "Sign" brings me to "signing a transaction" very quickly, in this context, but that's not what's happening here at all.   How about `"ApproveProposal"`?

Reply:
I renamed it to SignOff. I think SignOff has stronger implications than just an approval and better describes this action